Added entries for 18-623 + 18-723

b-joel-gonzalez opened the pull request:
# Summary of Changes

- Please indicate the classes you are modifying:
  - 18-623 (new)
  - 18-723 (new)

# Checklist:

Make sure that your contributions:

- [ ] Are accurate and will not mislead the reader
- [ ] Do not encourage or cause academic integrity violations
- [ ] Do not offend professors or unfairly criticize the university

b-joel-gonzalez committed:
Created 18623 + 18723

Update 18623.md

Update 18623.md

Update 18623.md

Update 18623.md

Create 18723

Rename 18723 to 18723.md

Update README.md

Update 18723.md

Update 18623.md

Update 18723.md

Update 18723.md

Update 18723.md

Update 18723.md

Update 18723.md

Update electives/18623.md

Update electives/18623.md

Update electives/18723.md

Co-Authored-By: Michael You <youmichaelc@gmail.com>

mikinty commented on electives/18623.md:
```diff
@@ -0,0 +1,66 @@
+# 18-623: Analog Integrated Circuit Design
```
Great class, glad you covered it :)

mikinty commented on electives/18623.md:
```diff
@@ -0,0 +1,66 @@
+
+The TAs will hold lab sessions on Fridays, some of which are dedicated to helping you out with the homework. Be sure to attend those, and come with lots of questions! That is a great time to get some clarification for the homework, as well as lecture concepts in general.
+
+Ultimately, the homeworks have a lower weight on your grade, relative to the projects and exams. However, you should still give a good effort on them - be sure to go back and learn what you did not understand fully when doing the homework. This will help you when it comes time to tackle the projects and exams.
```
```suggestion
Ultimately, the homeworks have a lower weight on your grade, relative to the projects and exams. However, you should still give a good effort on them -- be sure to go back and learn what you did not understand fully when doing the homework. This will help you when it comes time to tackle the projects and exams.
```

mikinty commented on electives/18623.md:
```diff
@@ -0,0 +1,66 @@
+
+## Exams
+
+There are two exams in the course. The first exam covers roughly the first ten lectures, and the second exam covers roughly the next ten lectures (though the second exam depends upon material from the first).
+
+The exams are very reflective of what is covered in the lectures and on the homeworks. As you go through each lecture, be sure to take notes on the different examples covered. Additionally, be sure to spend time reviewing the homeworks after you have completed them. Both will help prepare you for what is covered on the exams.
+
+The exams are similar in weight as the projects for your grade, so it is advised to prepare well for them.
```
From what I remember, the textbook by Razavi has lots of good problems that are similar / appear on the exam. A hidden gem that is often ignored in this class (and a lot of ECE classes since profs don't explicitly require you to read the textbook)

mikinty replied:
I also remember there are some questions on the exam that are quite computation heavy, so without good V=IR fundamentals or practice solving particular circuit questions / using certain assumptions, the problems can be quite unsolvable in the allotted time (50 min, right?). So practice is paramount.

b-joel-gonzalez replied:
Good points! I added in a bit about using his book for exam prep + reviewing circuit fundamentals before exams

mikinty commented on electives/18623.md:
```diff
@@ -0,0 +1,66 @@
+
+## Projects
+
+The projects are the "meat" of this course. Through these projects, you will learn how to use Cadence for circuit simulation and layout, as well as how to approach an analog IC design task.
```
From someone who took this class without prior Cadence experience. I suffered. 

So maybe add something in about how if you have 0 Cadence experience, it's a good idea to go to OH a lot to get familiar with it, or just take another class (421?) that uses Cadence.

b-joel-gonzalez replied:
18-421 is actually crosslisted with 18-623! So they're the same course now (not sure when they were crosslisted). I think 18-320 introduces a bit of Cadence before 421/623? But I don't think it's too much experience, from what I've heard... I added in a bit to that section to suggest that people spend time getting used to Cadence

mikinty commented on electives/18623.md:
```diff
@@ -0,0 +1,66 @@
+## Projects
+
+The projects are the "meat" of this course. Through these projects, you will learn how to use Cadence for circuit simulation and layout, as well as how to approach an analog IC design task.
+Consequently, you will likely spend a significant amount of time on the projects - but hopefully, you will be learning through doing. You will be asked to meet a number of design specifications for both projects; the challenge comes in meeting all specifications, since there are often key tradeoffs in design.
```
```suggestion
Consequently, you will likely spend a significant amount of time on the projects -- but hopefully, you will be learning through doing. You will be asked to meet a number of design specifications for both projects; the challenge comes in meeting all specifications, since there are often key tradeoffs in design.
```

mikinty commented on electives/18623.md:
```diff
@@ -0,0 +1,66 @@
+The projects are the "meat" of this course. Through these projects, you will learn how to use Cadence for circuit simulation and layout, as well as how to approach an analog IC design task.
+Consequently, you will likely spend a significant amount of time on the projects - but hopefully, you will be learning through doing. You will be asked to meet a number of design specifications for both projects; the challenge comes in meeting all specifications, since there are often key tradeoffs in design.
+
+The projects are done in pairs of two. Since you are working in Cadence, it can be tricky to divide responsibilities, but you should still be able to break the projects into subcomponents that you can both work on and then integrate.
```
I wish I had pairs when I took the class :(

mikinty commented on electives/18623.md:
```diff
@@ -0,0 +1,66 @@
+
+You will have a few short lab sessions that will catch you up on the basics of Cadence simulation and layout. But you'll really start to become familiar with its tools through working on the projects.
+Note that the projects are done over roughly the course of one month each. This may sound like a lot of time, but you will also have homeworks and exams in the course during this time - the earlier you start on the projects, the better!
+
```
Is it also worth mentioning about how to access Cadence? Using lab computers is the best way to do it, and it's so much faster than using a VM. I used VMs because I was often on the road (interviewing, traveling), and it was a PAIN to do anything. Might be worth mentioning that this is not a very remote-friendly class (only if the VM Cadence experience is still bad)

b-joel-gonzalez replied:
They actually have it so that we can remote access into the ECE clusters, and run the Cadence software remotely from our computer! That said, it can still be a bit laggy, especially on slower computers. I added a section suggesting that folks try setting up the remote access early, or just use the physical computers in the labs

mikinty commented on electives/18623.md:
```diff
@@ -0,0 +1,66 @@
+Additionally, some of the lectures from the middle of the class actually step through how to design an op-amp to meet certain specifications. You should definitely use these slides as an aid in designing your own op-amp for this project.
+
+Similar to the previous project, work with your teammate to split up portions of the design, test them individually, and then integrate together. You will also need to do layout for this project, but that should be saved for after you have finalized your simulations.
+
```
So I was not a great student in this class, but one thing I remember was that a lot of labs just felt like randomly guessing parameters / topologies and hoping it would work. And sometimes you would just go to OH and the TA would just magically come up with a topology for you to try and get to work.

Is there a better way to come up with / design the circuits, or is it really just guessing? Curious if you have any insight on this.

b-joel-gonzalez replied:
I struggled a lot on the projects :D so unfortunately I don't think I have the best advice besides telling folks to go to office hours to discuss their designs with the TAs. At least from my experience, the TAs were very helpful in walking me through my design and pointing me in the right direction. Some of the designs still seemed a bit mysterious (particularly for project 1) but OH definitely helped! I added a bit to talk about that

mikinty commented on electives/18623.md:
```diff
@@ -0,0 +1,66 @@
+- [Course Description](https://courses.ece.cmu.edu/18623)
+- Course textbook: _Analysis and Design of Analog Integrated Circuits_, 5th Edition
+- Additional text that may be useful: _Design of Analog CMOS Integrated Circuits_, 2nd Edition
+- [Professor Behzad Razavi's lectures](https://www.youtube.com/playlist?list=PLyYrySVqmyVPzvVlPW-TTzHhNWg1J_0LU): this playlist contains lectures on a number of topics that are relevant to 18623. Prof. Razavi has developed some amazing resources for learning IC design.
```
These are amazing, thanks for including these

mikinty commented on electives/18723.md:
```diff
@@ -0,0 +1,56 @@
+
+## Homework
+
+There is only a single homework for the class, and no exams. The homework is in the first few weeks of the course, and correspondingly focuses on concepts in those lectures. It does not involve any simulation - only mathematical analysis. Hopefully, it should not take too long, but be sure to go to office hours if you need to review those concepts.
```
```suggestion
There is only a single homework for the class, and no exams. The homework is in the first few weeks of the course, and correspondingly focuses on concepts in those lectures. It does not involve any simulation -- only mathematical analysis. Hopefully, it should not take too long, but be sure to go to office hours if you need to review those concepts.
```

mikinty reviewed (changes requested):
15 commits...is it possible to squash them down to maybe 3 or 2?

Thank you for adding these hardware classes. It's rare to see people get to these classes these days, especially with huge paychecks waiting at the FANG companies

mikinty commented:
Also sorry if I was a little picky with 18-623, just got excited to see a class that I've been in before (and struggled in!)

mikinty commented:
Wondering if you were planning to address the rest of the comments? I can also help contribute some to this as well

b-joel-gonzalez committed:
Updated 18623 + 18723

b-joel-gonzalez commented:
Hey, sorry about the delay! I saw your initial comments on my pull request and addressed some of them, but didn't get around to the rest of them till now.

I squashed the commits down to two commits, and I worked in all of your suggested changes (made comments for each one above). I also added in a section for both classes on Prerequisites, since I figured that it could help students out.

Thanks for the feedback on my pull request - let me know if you have any more thoughts on what I've added!

mikinty reviewed (approved):
Thanks for all the changes, appreciate it.

Regarding 421...I was actually looking for the class where you basically build adders...I think it's [18-340](https://courses.ece.cmu.edu/18340) so oops on me.
